src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/disk_encryption.py

```diff
@@ -515,20 +515,26 @@ def _show_post_action_message(resource_group_name, vmss_name, maunal_mode, enabl
 
 
 def show_vmss_encryption_status(cmd, resource_group_name, vmss_name):
-    encryption_ext_names = [v['name'] for v in vmss_extension_info.values()]
-    views = get_vmss_instance_view(cmd, resource_group_name, vmss_name)
-    if not views.extensions or not [e for e in views.extensions if e.name in encryption_ext_names]:
-        from knack.util import CLIError
-        raise CLIError("'{}' is not encrypted yet".format(vmss_name))
-
-    views = get_vmss_instance_view(cmd, resource_group_name, vmss_name, instance_id='*')
-    result = [{'disks': v.disks, 'extensions': v.extensions} for v in views]
-    # get rid of unrelaed disk status
-    for r in result:
-        for d in r['disks']:
-            d.statuses = [s for s in d.statuses if s.code.startswith('EncryptionState')]
-        r['encryption-extension'] = next((e for e in (r['extensions'] or []) if e.name in encryption_ext_names), None)
-        r.pop('extensions')  # we don't need this array any more
+    client = _compute_client_factory(cmd.cli_ctx)
+    vm_instances = list(client.virtual_machine_scale_set_vms.list(resource_group_name, vmss_name,
+                                                                  select='instanceView', expand='instanceView'))
+    result = []
+    for instance in vm_instances:
+        view = instance.instance_view
+        disk_infos = []
+        vm_enc_info = {
+            'name': '',
```

Comment:
If name is always empty, is there a reason to include it?

Reply:
This is a good catch. The vm instance does have a `name` field, but always be `None`. To end users what really matters is the `id`. So I will just remove the `name` from the output, instead of strictly copying PS's logics